### PR TITLE
[RAINCATCH-514] Update readme with info about session storage feature

### PR DESCRIPTION
**Pull request by paolobueno**
# Motivation
Document a migration guide for utilizing session storage

# Changes
- Add 0.2.x session to readme
- Add jsdoc to mbaas router init

### CODE REVIEW

**Comment by paolobueno** [comment]
ping @nialldonnellyfh

File: lib/router/mbaas.js [hunk 1]
```diff
@@ -119,6 +119,14 @@ function trimAuthResponse(authResponse, exclusionList) {
...
 }
 
+/**
+ * Initializes the router, mounting it in the supplied express application
+ * @param  {Mediator}   mediator                  Mediator instance from fh-wfm-mediator
+ * @param  {Express.App}   app                       Express application
+ * @param  {Array}   authResponseExclusionList List of fields in the User schema to exclude from responses
+ * @param  {Object}   sessionOptions            Options for storage and express-session
+ * @param  {Function} cb                        Node-style callback
+ */
```
Comment (ghost) [inline lib/router/mbaas.js:129]: align spaces in jsdoc

**Comment by ghost** [comment]
LGTM

File: README.md [hunk 2]
```diff
@@ -4,6 +4,50 @@ A module for FeedHenry RainCatcher that manages users, groups and memberships. I
 - Backend services to handle CRUD operations for user, group and membership.
 - Frontend directives and services providing CRUD clients for user, group and membership.
 
+## Upgrading to 0.2.0 from 0.1.x
+Version 0.2.0 introduces session storage for authenticated users utilizing either mongodb or redis as backends.
```
Comment (nialldonnellyfh) [inline README.md]: - Can you capitalise MongoDB and Redis.
- Possibly better to call them "backend stores" or "storage engines" ? WDYT

File: README.md [hunk 3]
```diff
@@ -4,6 +4,50 @@ A module for FeedHenry RainCatcher that manages users, groups and memberships. I
...
+
+This involves an extra parameter to the initialization of the router for the authentication service which contains the configuration for the session storage.
+
+### How to upgrade
+
+In the auth-service:
```
Comment (nialldonnellyfh) [inline README.md]: Maybe: In the MBaaS service that authenticates users (e.g. raincatcher-demo-auth), initialise the session store with the configuration shown below.

**Review by nialldonnellyfh — COMMENTED** [review]
The README also needs information on securing endpoints with the session middlware in the raincatcher-demo-cloud-app

E.g. https://github.com/feedhenry-raincatcher/raincatcher-demo-cloud/pull/33/files

File: README.md (hunk 3 above)
Comment (paolobueno) [reply README.md]: Going to go with your text, I was thinking about using terms not tied in with RHMAP (i.e. MBaaS service), but I guess that's better in our context.

File: README.md (hunk 2 above)
Comment (paolobueno) [reply README.md]: Done!

File: lib/router/mbaas.js (hunk 1 above)
Comment (paolobueno) [reply lib/router/mbaas.js:129]: Done, thanks!

**Comment by paolobueno** [comment]
Squashing and merging to feature branch

File: README.md [hunk 4]
```diff
@@ -4,6 +4,50 @@ A module for FeedHenry RainCatcher that manages users, groups and memberships. I
...
+      secret: 'raincatcher',
+      resave: false,
+      saveUninitialized: true,
+      cookie: {
+        secure: true,
+        httpOnly: true,
+        path: '/'
+      }
+      // `url` is used by the 'mongo' store for configuration
+      url: 'mongodb://localhost:27017/raincatcher-demo-auth-session-store'
```
Comment (ghost) [inline README.md:39]: shouldnt we get url from ENVs(i.e. `url: process.env.FH_MONGODB_CONN_URL`)?

File: README.md [hunk 5]
```diff
@@ -4,6 +4,50 @@ A module for FeedHenry RainCatcher that manages users, groups and memberships. I
...
+      // 'host' and 'port' are used by the 'redis' store
+      host: '127.0.0.1',
```
Comment (ghost) [inline README.md:41]: same for ENVs for redis(i.e. `process.env.FH_REDIS_HOST`)?

File: README.md (hunk 4 above)
Comment (paolobueno) [reply README.md:39]: That's the application/client code's responsibility. In practice we do that in demo auth here: https://github.com/feedhenry-raincatcher/raincatcher-demo-auth/blob/feature/persistent-storage/lib/sessionInit.js
The `$fh.db` call will grab the url from `FH_MONGODB_CONN_URL`

**Pull request merged**